source/blog/2018-06-29-ember-3-2-released.md

```diff
@@ -0,0 +1,295 @@
+
+```javascript
+Logger.debug(message);
+```
+
+into this:
+
+```javascript
+if (console.debug) {
+  console.debug(message);
+} else {
+  console.log(message);
+}
+```
+
+**Note for Add-on Authors** - If your add-on needs to support both Ember 2.x and Ember 3.x clients, you will need to test for the existence of console before calling its methods. If you do much logging, you may find it convenient to define your own wrapper. Writing the wrapper as a service will provide for dependency injection by tests and perhaps even clients.
+
+##### Private API Router#router renamed (2 of 3)
+
+In order to avoid collisions with user-defined properties or methods, the `Router#router` private API has been renamed to `Router#_router`. If you want access to the router, you should inject the router service into the route like this:
+
+```javascript
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default Route.extend({
+  router: service()
+});
+```
+
+##### Define computed properties with defineProperty (3 of 3)
+
+Use `defineProperty` (via `import { defineProperty } from '@ember/object';`) to define computed properties. Although uncommon, it is possible to assign computed properties directly to objects. This way they are implicitly computed from things like `Ember.get`. Assigning computed properties directly is deprecated to support ES5 getter computed properties, and you should replace these assignments with calls to `defineProperty`.
+
+For example, the following:
+
+```javascript
+import { computed, get } from '@ember/object';
+let object = {};
+object.key = computed(() => 'value');
+get(object, 'key') === 'value';
+```
+
+Should be changed to this:
+
+```javascript
+import { computed, defineProperty, get } from '@ember/object';
+let object = {};
+defineProperty(object, 'key', computed(() => 'value'));
+get(object, 'key') === 'value';
+```
+
+For more details on changes in Ember.js 3.2, please review the [Ember.js 3.2.0 release page](https://github.com/emberjs/ember.js/releases/tag/v3.2.0).
+
+---
+
+## Ember Data
+
+Ember Data is the official data persistence library for Ember.js applications.
+
+### Changes in Ember Data 3.2
+
+There are four (4) new features and zero (0) deprecations in the Ember Data 3.2 release.
+
+#### New Features (4)
+
+##### Lazy Relationship Payloads (1 of 4)
+
+In Ember Data 2.14 lazy-relationship parsing was introduced. Because this parsing used left-side/right-side keying, this meant it was not compatible with polymorphic relationships.
+
+With Ember Data 3.2 this is [now fixed](https://github.com/emberjs/data/pull/5230).
+
+##### Ember Data Feature Flag Removal (2 of 4)
+
+The current feature flags for Ember Data has gone stale, therefore they are all [removed in Ember Data 3.2](https://github.com/emberjs/data/pull/5384). Ember Data is going to attempt to go in a different direction with some of the planned changes for 2018.
```

Comment:
I think it would be 

```
[...] feature flags for Ember Data have gone stale, [...]
```

Reply:
I'm also wondering what a stale feature flag describes. Does this mean that the feature flag hid a feature for a long time that is not going to make it into a stable release in the end?

I'm not sure if it's just me though - possibly most readers are familiar with that term already. If that's the case, we could write more about it in another, more casual medium, e.g. The Ember Times

cc @amyrlam @kennethlarsen

Reply:
Great idea- maybe runspired could help illuminate?